### PR TITLE
feat: Skill point loadouts

### PR DESCRIPTION
**Pull request by kristofbolyai**
This is @DonkeyBlaster's PR, but squashed. Hopefully spotless won't be triggered for all the merge commits adding 2024 headers..

### CODE REVIEW

File: common/src/main/java/com/wynntils/models/character/SkillPointModel.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/character/SkillPointModel.java:144]: This should be okay for now.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java]: I think magicus requested that you make `isBuild` a method, basically revert it to as it was. Sorry for the confusion.

File: common/src/main/java/com/wynntils/screens/skillpointloadouts/SkillPointLoadoutsScreen.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/skillpointloadouts/SkillPointLoadoutsScreen.java]: I think I previously commented on how these buttons should be separate classes, not anonymous ones. I think you only converted the first one, and forgot the others.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java]: These methods are not needed.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java:23]: Should be List.of() so it is an unmodifiable list

File: common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java:30]: Missed this button too.

Also, do these buttons need to be separate classes, I think you can just use the button builder as from what I can tell none of the buttons do anything different from a regular button

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java:30]: This button specifically can be in the feature class, as it's injected by the feature, and not used on a custom screen. 

Do you want to use button builders in regular screens? Sure, we did that before, if there is nothing special about them.

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java:30]: I think using the builder is fine, for example the load button can just be this and (I think) it would work exactly the same, no need for a seperate LoadButton class

```
loadButton = new Button.Builder(
                        Component.translatable("screens.wynntils.skillPointLoadouts.load"),
                        (b) -> Models.SkillPoint.loadLoadout(selectedLoadout.key()))
                .pos((int) (dividedWidth * 35), (int) (dividedHeight * 52))
                .size((int) ((dividedWidth * 44) - (dividedWidth * 35)), BUTTON_HEIGHT)
                .build();
```

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java:23]: Doesn't work with gson for some reason, when I try to save anything with List.of() it does not persist after relaunch

**Comment by kristofbolyai** [comment]
`[20:25:00] [Render thread/WARN] (wynntils) Skill Point Model failed to parse tome: StyledText{' §6Requirements: - §a✔ §7Lv. Min: 105§6- §c✖ §7Raid Req: §f10§7/15'}`

Some overlooked error, should be easily to handle.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java:23]: Yeah, that might be the case. I'll just make sure to support it separately in a new PR eventually. Just use the immutable wrappers here for now, and i'll handle proper serialization support later.

File: common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java:30]: Hm, it very well could be changed, it is a lot less lines, so maybe it's cleaner. Although having a separate class has it's cleanness too. Let's leave it up to DonkeyBlaster to decide.

Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/features/utilities/SkillPointLoadoutsFeature.java:30]: I will change it to a builder, having a separate class in a non-screen feels like a waste

oh wait no the actual convert/delete/loads I will keep as separate since they are screen parts, if the LoadoutScreenButton class is fine in the feature I will leave it

**Review by kristofbolyai — COMMENTED** [review]
Looks good now! I'll merge this PR with the chat item one, when the time comes.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (magicus) [inline common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java:52]: The traditional concept of a "build" in Wynncraft has also included a weapon. I'm not demanding a change in this PR, but maybe we should think of how this could be implemented in the future. That would make it more compatible with e.g. wynndata.

**Review by magicus — APPROVED** [review]
I've looked through this, some parts more cursory and other in depth. I think it looks good.

File: common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java
Comment (ryanzhoudev) [reply common/src/main/java/com/wynntils/models/character/type/SavableSkillPointSet.java:52]: Yup, the eventual goal is to integrate with bolyai's gear encoding standard and have exporting to/from wynnbuilder
(but the legacy portion is working already, so why not)

**Pull request merged**
